### ISSUE TITLE
Ref indexing not allowed as initializer value 

### ISSUE THREAD

**Issue by cyborgar** [irmen/prog8#41]
This may be fixed in the latest source but I thought I should report this just in case. The following doesn't work

```
main {
  ubyte[] bytearr = [0,1,2,3,4]
  
  sub start() {
    uword ref = &bytearr
    ubyte get = ref[0]
  }
}
```

I get 

```
* internal error *
Exception in thread "main" java.lang.IllegalArgumentException: Failed requirement.
        at prog8.ast.statements.VarDecl.replaceChildNode(AstStatements.kt:235)
        at prog8.ast.walk.IAstModification$ReplaceNode.perform(AstWalker.kt:62)
        at prog8.ast.walk.AstWalker.applyModifications(AstWalker.kt:178)
        at prog8.compiler.astprocessing.AstExtensionsKt.reorderStatements(AstExtensions.kt:26)
        at prog8.compiler.CompilerKt.processAst(Compiler.kt:252)
        at prog8.compiler.CompilerKt.compileProgram(Compiler.kt:97)
        at prog8.CompilerMainKt.compileMain(CompilerMain.kt:106)
        at prog8.CompilerMainKt.main(CompilerMain.kt:23)
```

I've attached a small test program for testing

[test.zip](https://github.com/irmen/prog8/files/6472660/test.zip)

**Comment by irmen** [comment]
yeah this was fixed shortly after 6.5-beta so for example the 7.0-beta release should compile this just fine.

**Comment by cyborgar** [comment]
Yes, I suspected this was fixed.
